tests/database.test.js

```diff
@@ -0,0 +1,139 @@
+const app = require('../server/app.js');
+const supertest = require("supertest");
+const http = require('http');
+const mongoose = require('mongoose');
+const { User, Course, SeatingLayout } = require('../server/dbSchemas/attendanceSchema.js');
+
+describe('Database Functionality', () => {
+
+    let request;
+    let server;
+
+    // Open database before running tests
+    beforeAll(async (done) => {
+        server = http.createServer(app);
+        server.listen();
+        request = supertest(server);
+        const db = mongoose.connection;
+        db.once('open', done);
+    });
+
+    // Close database & server when done
+    afterAll(async (done) => {
+        await mongoose.connection.close();
+        await server.close(done);
+    });
+
+    it('Should store user to database', async done => {
+        //Send request to register user
+        const res = await request.post('/api/register').send({
+            email: 'dummy@test.com',
+            name: 'dummy test',
+            password:'12345',
+            is_professor: true
+        });
+
+        //Search user in database by email
+        const user = await User.findOne({
+            email: 'dummy@test.com'
+        });
+
+        //Check that user has name, email, password, and is or is not a professor
+        expect(user.name).toBeTruthy()
+        expect(user.email).toBeTruthy()
+        expect(user.password).toBeTruthy()
+        expect(typeof(user.is_professor) === 'boolean').toBeTruthy();
+
+        //Delete test user after testing
+        response = await request.delete("/api/delete-user").send({
+            email: 'dummy@test.com'
+        });
+
```

Comment:
Here we might want to check that the delete request worked (i.e. we want to know if it fails)
```suggestion
        expect(response.status).toBe(200);
```
**Note**: We might also want to update the other tests which also make delete requests so that they also make sure their request succeeds.